### PR TITLE
Helm: Support of extraVolumeClaimTemplates for writer statefulset

### PR DESCRIPTION
**Pull request by anosulchik**
**What this PR does / why we need it**:

This PR adds `extraVolumeClaimTemplates` value for write loki component, it allows to have more than one volume claims for persistent statefulset configuration. For example, when it's needed to have ingester's and shipper's WAL files on different persistent volumes. 

**Special notes for your reviewer**:

**Checklist**
- [X] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [X] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/helm/loki/values.yaml [hunk 1]
```diff
@@ -678,6 +678,8 @@ write:
   extraVolumeMounts: []
   # -- Volumes to add to the write pods
   extraVolumes: []
+  # -- volumeClaimTemplates to add to statefulset
```
Comment (JStickler) [inline production/helm/loki/values.yaml:681]:
```suggestion
  # -- volumeClaimTemplates to add to StatefulSet
```

**Review by JStickler — APPROVED** [review]
[Docs team]  Kubernetes capitalizes as StatefulSet.  Other than that, LGTM.

**Comment by anosulchik** [comment]
@MichelHollands I've updated changelog and version. All tests are passed. Thank you.

**Comment by anosulchik** [comment]
Guys, would you please review my recent changes and merge if all is OK? Thanks

**Review by MichelHollands — APPROVED** [review]

**Pull request merged**